extensions/ql-vscode/src/data-extensions-editor/auto-model-codeml-queries.ts

```diff
@@ -87,30 +73,89 @@ export async function runAutoModelQueries({
     await cliServer.resolveQlpacks(additionalPacks, true),
   );
 
-  const candidates = await runAutoModelQuery({
-    mode,
-    queryTag: "candidates",
+  // Run the actual query
+  const completedQuery = await runQuery({
     cliServer,
     queryRunner,
     databaseItem,
-    qlpack,
-    sourceInfo,
+    queryPath,
+    queryStorageDir,
     additionalPacks,
     extensionPacks,
-    queryStorageDir,
     progress,
     token: cancellationTokenSource.token,
   });
 
-  if (!candidates) {
+  if (!completedQuery) {
     return undefined;
   }
 
+  // Get metadata for the query. This is required to interpret the results. We already know the kind is problem
+  // (because of the constraint in resolveQueries), so we don't need any more checks on the metadata.
+  const metadata = await cliServer.resolveMetadata(queryPath);
+
+  // CodeQL needs to have access to the database to be able to retrieve the
+  // snippets from it. The source location prefix is used to determine the
+  // base path of the database.
+  const sourceLocationPrefix = await databaseItem.getSourceLocationPrefix(
+    cliServer,
+  );
+  const sourceArchiveUri = databaseItem.sourceArchive;
+  const sourceInfo =
+    sourceArchiveUri === undefined
+      ? undefined
+      : {
+          sourceArchive: sourceArchiveUri.fsPath,
+          sourceLocationPrefix,
+        };
+
+  const candidates = await interpretAutomodelResults(
+    cliServer,
+    completedQuery,
+    metadata,
+    sourceInfo,
+  );
+
   return {
     candidates,
   };
 }
 
+async function resolveAutomodelQueries(
```

Comment:
Since we only expect 1 query and throw if we don't resolve exactly 1 query, I think this function should be called `resolveAutomodelQuery`